toxics/timeout_test.go

```diff
@@ -0,0 +1,117 @@
+package toxics_test
+
+import (
+	"io"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/Shopify/toxiproxy/toxics"
+)
+
+func TestTimeoutToxicDoesNotCauseHang(t *testing.T) {
+	ln, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatal("Failed to create TCP server", err)
+	}
+	defer ln.Close()
+
+	proxy := NewTestProxy("test", ln.Addr().String())
+	proxy.Start()
+	defer proxy.Stop()
+
+	proxy.Toxics.AddToxicJson(ToxicToJson(t, "might_block", "latency", "upstream", &toxics.LatencyToxic{Latency: 10}))
+	proxy.Toxics.AddToxicJson(ToxicToJson(t, "to_delete", "timeout", "upstream", &toxics.TimeoutToxic{Timeout: 0}))
+
+	serverConnRecv := make(chan net.Conn)
+	go func() {
```

Comment:
It almost seems at this point we need a helper for the pattern of starting a server and accepting connections, as it's somewhat common and should be encouraged.